### PR TITLE
Update: Elide time dimension filter value format

### PR DESCRIPTION
**Pull request by jkusa**
## Description

Update: Time dimension filter value format

## Proposed Changes

- Refactor time grains
  - Reorganize time grain and period types
  - Add `isoWeek` as grain
  - Normalize bard/fili `week` grain to `isoWeek`
  - Replace ISODatePeriod util methods
  - Remove get-date-range file
- Elide time dimension filter value format

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/adapters/facts/elide.ts [hunk 1]
```diff
@@ -55,39 +56,63 @@ export default class ElideFactsAdapter extends EmberObject implements NaviFactAd
...
-  private dataQueryFromRequest(request: RequestV2): string {
-    const args = [];
-    const { table, columns, sorts, limit, filters } = request;
-    const columnsStr = columns.map(col => getElideField(col.field, col.parameters)).join(' ');
+  private readonly grainFormats: Record<SupportedGrains, string> = {
+    second: 'yyy-MM-DDTHH:mm:ss',
+    minute: 'yyyy-MM-DDTHH:mm',
+    hour: 'yyyy-MM-DDTHH',
+    day: 'yyyy-MM-DD',
+    week: 'yyyy-MM-dd',
```
Comment (jkusa) [inline packages/data/addon/adapters/facts/elide.ts:64]:
```suggestion
    week: 'yyyy-MM-DD',
```

File: packages/data/tests/unit/utils/date-test.ts [hunk 2]
```diff
@@ -1,383 +1,229 @@
...
-      'getFirstDayOfPrevIsoDateTimePeriod should return: ' + expectedDate4
-    );
-  });
-
-  test('getFirstDayEpochIsoDateTimePeriod - Epoc date as mid of DateTimePeriod', function(assert) {
-    assert.expect(3);
-
-    let dateFormat = 'YYYY-MM-DD',
-      originalEpoch = config.navi.dataEpoch;
+  test('getFirstDayEpochForGrain - Epoc date as mid of DateTimePeriod', function(assert) {
```
Comment (kevinhinterlong) [inline packages/data/tests/unit/utils/date-test.ts:14]:
```suggestion
  test('getFirstDayEpochForGrain - Epoch date as mid of DateTimePeriod', function(assert) {
```

File: packages/core/addon/helpers/format-date-for-granularity.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ export function formatDateForGranularity(date, granularity) {
     return '--';
   }
 
-  if (granularity === 'week') {
+  if (granularity === 'week' || granularity === 'isoWeek') {
```
Comment (kevinhinterlong) [inline packages/core/addon/helpers/format-date-for-granularity.js:53]: Could check if the period is week instead of using grain

**Review by kevinhinterlong — APPROVED** [review]
Aside from potential issues with the standard 'week' grain, 👍

**Pull request merged**